scripts/generate-conda-envs.py

```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2020-2022, NVIDIA CORPORATION. All rights reserved.
+#
+# NVIDIA CORPORATION and its licensors retain all intellectual property
+# and proprietary rights in and to this software, related documentation
+# and any modifications thereto.  Any use, reproduction, disclosure or
+# distribution of this software and related documentation without an express
+# license agreement from NVIDIA CORPORATION is strictly prohibited.
+#
+# See the LICENSE file for details.
+#
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Literal, Protocol, Tuple
+
+from jinja2 import Template
+from typing_extensions import TypeAlias
```

Comment:
Is it practical to only use standard library modules in this script? If we're asking users to run this manually to create an env file to get their dependencies, then ideally I wouldn't want to assume that they have anything installed except a vanilla installation of python 3.

Reply:
for sure we can dispense with `TypeAlias` in this script

Reply:
Also `jinja2` templates, which might be a bigger ask.

Reply:
It's doable, it will just be tedious

Reply:
https://github.com/nv-legate/legate.core/pull/367/commits/c42bee8ef46343508f367a47ee3a405701c8261d